src/FilterButtonList.js

```diff
@@ -0,0 +1,139 @@
+import F from 'futil-js'
+import _ from 'lodash/fp'
+import React from 'react'
+import { observer } from 'mobx-react'
+import {
+  Dynamic,
+  Flex,
+  CheckButton as DefaultCheckButton,
+  Modal as DefaultModal,
+  Popover as DefaultPopover,
+} from './layout'
+import DefaultIcon from './DefaultIcon'
+import InjectTreeNode from './utils/injectTreeNode'
+import styles from './styles'
+
+export let DefaultMissingTypeComponent = InjectTreeNode(({ node = {} }) => (
+  <div>
+    Type <b>{node.type}</b> is not supported (for key <i>{node.key}</i>)
+  </div>
+))
+
+let FieldDescription = observer(({ node }) => (
+  <div className="filter-description">
+    {node.fieldDescription} {node.typeDescription}
+  </div>
+))
+
+let FilterButtonItem = ({
+  node,
+  tree,
+  fields,
+  mapNodeToProps,
+  label,
+  Button,
+  CheckButton,
+  Modal,
+}) => {
+  let modal = F.stateLens(React.useState(false))
+  let title =
+    label || _.get([node.field, 'label'], fields) || node.field || node.key
+  return (
+    <>
+      <CheckButton checked={node.hasValue} onClick={F.on(modal)}>
+        {title}
+      </CheckButton>
+      <Modal isOpen={modal}>
+        <div className="filter-button-modal">
+          <h1>{title}</h1>
+          <FieldDescription node={node} />
+          <div className="filter-component">
+            <Dynamic
+              tree={tree}
+              node={node}
+              path={_.toArray(node.path)}
+              {...mapNodeToProps(node, fields)}
+            />
+          </div>
+          <Button onClick={() => tree.clear(node.path)}>Clear</Button>
+          <Button primary onClick={F.off(modal)}>
+            Done
+          </Button>
+        </div>
+      </Modal>
+    </>
+  )
+}
+
+let Box = ({ nodeJoinColor, children, nested, className }) => (
+  <Flex
+    wrap
+    className={`${className} ${nested ? 'nested' : ''}`}
+    alignItems="center"
+    style={{ borderColor: nodeJoinColor }}
+  >
+    {children}
+  </Flex>
+)
+
+let FilterButtonList = observer(
+  ({
+    node,
```

Comment:
if this is not using `injectTreeNode`, are we expecting this to be explicitly passed from the client? I'm ok with it in theory, but we do have precedent of supporting `path` + `tree` instead of requiring end users to do `node={tree.getNode(path)}`